infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -234,11 +235,19 @@ public void setupTopologyConfigId(long topologyConfigId) {
             IRetry.build(IntervalRetry.class, () -> {
                 try (TxnContext txn = corfuStore.txn(NAMESPACE)) {
                     for (LogReplicationMetadataType type : LogReplicationMetadataType.values()) {
-                        long val = Address.NON_ADDRESS;
                         if (type == LogReplicationMetadataType.TOPOLOGY_CONFIG_ID) {
-                            val = topologyConfigId;
+                            appendUpdate(txn, type, topologyConfigId);
+                        } else if (type == LogReplicationMetadataType.VERSION) {
+                            String version = LogReplicationStreamNameTableManager.getCurrentVersion();
+                            if (version == null) {
+                                log.warn("Failed to fetch version from plugin.");
```

Comment:
I think this must be logged as an error.

Reply:
Changed, thanks!